assets/js/modules/analytics-4/datastore/webdatastreams.test.js

```diff
@@ -274,14 +274,59 @@ describe( 'modules/analytics-4 webdatastreams', () => {
 
 		describe( 'getMatchingWebDataStream', () => {
 			const webDataStreams = [ webDataStreamDotCom, webDataStreamDotOrg ];
+
+			it( 'should return NULL when no datastreams are matched', () => {
+				provideSiteInfo( registry, {
+					referenceSiteURL: 'http://example.net',
+				} );
+
+				const datastream = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getMatchingWebDataStream( webDataStreams );
+
+				expect( datastream ).toBeNull();
+			} );
+
+			it( 'should return the correct datastream when reference site URL matches exactly', () => {
+				provideSiteInfo( registry, {
+					referenceSiteURL: 'http://example.com',
+				} );
+
+				const datastream = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getMatchingWebDataStream( webDataStreams );
+
+				expect( datastream ).toEqual( webDataStreamDotCom );
+			} );
```

Comment:
Previously we had a flaw which caused this to appear to work only because it was assuming the first web data stream for a property was a match. In order to make sure this is not the case, please add another test to make sure the same works when the matching data stream is not the first. This is covered below implicitly, but let's cover it explicitly due to the significance here.

Reply:
Good call. I've updated the relevant test cases to match the first and the second/last property. Let me know if this looks good. Thank you!